web/modules/custom/usa_orphaned_entities/usa_orphaned_entities.module

```diff
@@ -0,0 +1,94 @@
+<?php
+
+use Drupal\node\Entity\Node;
+
+/**
+ * @file
+ * Auto-delete orphaned synonyms and language toggles when deleting
+ * Federal Directory Records.
+ */
+/**
+ * Implements hook_form_alter.
+ */
+
+function usa_orphaned_entities_form_alter(&$form, &$form_state, $form_id) {
+  if (str_contains($form_id, 'delete_form')) {
+    $markup = "<div class=\"default-message\">" .
+              $form_state->getFormObject()->getDescription() .
+              "</div>";
+    $entity = $form_state->getFormObject()->getEntity();
+    $segway = "T";
+    $additional = "";
+
+    if ($form_id == 'node_directory_record_delete_form') {
+      $deleted_id = $entity->id();
+      $deleted_type = $entity->getEntityTypeId();
+
+      if ($deleted_type == "node") {
+        $query = \Drupal::entityQuery('node')
+          ->condition('type', 'agency_synonym')
+          ->condition('field_agency_reference', $deleted_id)
+          ->accessCheck(TRUE);
+        $results = $query->execute();
+        $synonyms = Node::loadMultiple($results);
+
+        if (!empty($synonyms)) {
+          $additional = "<br />" . PHP_EOL;
+          $additional .= "<div class=\"synonym-delete\">";
+          $additional .= "<span>Deleting this node will also delete the following nodes:</span>" . PHP_EOL;
+          $additional .= "<ul>" . PHP_EOL;
+          foreach ($synonyms as $synonym) {
+            $additional .= "<li>Synonym: <a href=\"/node/" . $synonym->id() . "\" target=\"_blank\">" . $synonym->title->value . " (" . $synonym->id() . ")</a></li>" . PHP_EOL;
+          }
+          $additional .= "</ul></div>" . PHP_EOL;
+          $segway = "Additionally, t";
+        }
+      }
+    }
+
+    if (!empty($entity->field_language_toggle->target_id)) {
```

Comment:
I created two "basic page" nodes, set the toggle on node A to node B, and when I deleted node B, I didn't get a warning about node A's toggle, nor was it deleted. And I think this line is why -- you're checking for a toggle on the node that's being deleted. 

Usually, for a pair of pages, toggles will be created in both directions, but that won't always be the case (usually because someone forgot a step). It would also be possible (I guess) for there to be multiple nodes with toggles pointing to the same node. So can we do a search based on that? 

I'm sure this is part of the reason Drupal doesn't do this automatically...

Reply:
This is def an oversight, I didn't consider this possibility.  Thankfully, it should be rather easy to find entities with any given NID and clear them.  My only worry is memory at the moment, but we'll see.